gcc-4.1 on sparc64 appears to miscompile the expression, maybe load the last
value even though the former is already NULL - I'll check in-depth some time
later when I have more time.

  - Rene Rebe <rene@exactcode.de>

--- embutils-0.17/ls.c	2006-06-18 11:25:22.170000000 +0000
+++ ./ls.c	2006-06-18 12:00:50.270000000 +0000
@@ -733,8 +733,12 @@
     char *expanded;
     if (e->d_name[0]=='.') {
       /* is it "." or ".."? */
-      if (e->d_name[1]==0 || (e->d_name[1]=='.' && e->d_name[2]==0))
+      if (e->d_name[1] == 0) {
 	if (_A || !_a) continue;
+      } if (e->d_name[1] == '.') {
+          if(e->d_name[2] == 0)
+            if (_A || !_a) continue;
+      }
       if (!_A && !_a) continue;
     }
     expanded=alloca(strlen(name)+strlen(e->d_name)+3);
